xcell/cls/cl.py

```diff
@@ -173,24 +180,35 @@ def get_cl_file(self):
                 nl_cp = mapper1.get_nl_coupled()
                 nl = w.decouple_cell(nl_cp)
             else:
-                nmap1 = 1 + (mapper1.get_spin() > 0)
-                nmap2 = 1 + (mapper2.get_spin() > 0)
-                nl_cp = np.zeros((nmap1 * nmap2, 3 * self.nside))
-                nl = np.zeros([nmap1 * nmap2, self.b.get_n_bands()])
+                n_cls = self.get_n_cls()
+                nl_cp = np.zeros((n_cls, 3 * self.nside))
+                nl = np.zeros([n_cls, self.b.get_n_bands()])
             # Signal
             if auto and mapper1.custom_auto:
                 cl_cp = mapper1.get_cl_coupled()
                 cl = w.decouple_cell(cl_cp)
+                cls_cov = mapper1.get_cls_covar_coupled()
+                cl_cov_cp = cls_cov['cross']
```

Comment:
Could you add a comment here about these new terms?

Reply:
OK, I've added an extensive comment explaining what this PR is supposed to implement.